src/services/SearchResults.cpp

```diff
@@ -103,36 +103,40 @@ static const char* ComparisonString(ComparisonType nCompareType)
     }
 }
 
-static unsigned int GetValue(const unsigned char* pBuffer, unsigned int nOffset, ComparisonVariableSize nSize)
+_Success_(return)
+_NODISCARD _CONSTANT_FN GetValue(_In_z_ const unsigned char* const pBuffer,
+                                 _In_   unsigned int nOffset,
+                                 _In_   MemSize nSize) noexcept
 {
+    // TBD: See if std::byte fits are purposes better
+    [[maybe_unused]] const auto uItemAtOffs{ static_cast<unsigned int>(pBuffer[nOffset]) };
```

Comment:
This feels like unnecessary code obfuscation. The duplication made it clear exactly what was happening in each case, and didn't require declaring the variable as sometimes unused.

Reply:
The return type is different in the first case. I'll try to do it with more clarity.